src/services/hsm.service.js

```diff
@@ -219,17 +219,33 @@ class HsmSession {
       });
     }
 
-    /* istanbul ignore next */
-    if (tx.version === hathorLib.constants.NANO_CONTRACTS_VERSION) {
+    /** @type {hathorLib.Address} */
+    let address = null;
+
+    if (tx.isNanoContract()) {
+      address = hathorLib.transactionUtils.getNanoContractCaller(tx);
+    }
+
+    if (tx.version === hathorLib.constants.ON_CHAIN_BLUEPRINTS_VERSION) {
       const { pubkey } = tx;
-      const address = hathorLib.addressUtils.getAddressFromPubkey(pubkey.toString('hex'), storage.config.getNetwork());
+      address = hathorLib.addressUtils.getAddressFromPubkey(pubkey.toString('hex'), storage.config.getNetwork());
+    }
+
+    if (address) {
       const addressInfo = await storage.getAddressInfo(address.base58);
       if (!addressInfo) {
-        // Not a wallet address
-        return { inputSignatures };
+        throw new Error(`No address info found for ${address.base58}`);
```

Comment:
question: Is that a breaking change from the expected behavior? Are there any cases that would expect a valid response with the `inputSignatures` that will now break instead?